### ISSUE TITLE
[codebuild] Is it possible to use BitBucket private repos as the source?

### ISSUE THREAD

**Issue by jpSimkins** [aws/aws-cdk#9654]
### The Question

Is it possible to use BitBucket for the source of CodeBuild? I have private repos and with Terraform, I was able to use CodeStar to get it to work. albeit, I had to do the initial OAuth connection myself and just use the ARN.

In the docs, I have seen:
```
const bbSource = codebuild.Source.bitBucket({
  owner: 'owner',
  repo: 'repo',
});
```
But this will not work for private repos. 

I have also attempted to use the codestar construct but it only works for GitHub. I haven't found a way to use the ARN to specify the resource.

### Environment
Language (Version):
  - **CDK CLI Version:** 1.57.0 (build 2ccfc50)
  - **Module Version:** 1.57.0
  - **Node.js Version:** v12.18.3
  - **OS:** Linux Zeus 5.4.0-42-generic #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
  - **Language (Version):**  typescript (3.7.5)

### Other information

Terraform allows me to use the ARN of codestar to use BitBucket private repos. Am I not able to with the CDK? So far, I have tried what the docs [suggest](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-codebuild-readme.html#bitbucketsource) but that only works for public repos. 

Ideally, I would like to see better integration into BitBucket. It would be nice to use the CodeStar construct to allow for BitBucket private repo access or at minimum, use the ARN for an existing resource.

**Comment by jonny-rimek** [comment]
you can also trigger codebuild from s3. e.g. having a webhook that triggers a lambda that copies the code to s3. not very nice, but better than no solution at all I guess^^

**Comment by skinny85** [comment]
Hey @jpSimkins ,

thanks for opening the issue.

> In the docs, I have seen:
> 
> ```
> const bbSource = codebuild.Source.bitBucket({
>   owner: 'owner',
>   repo: 'repo',
> });
> ```
> 
> But this will not work for private repos.

Why? Have you tried it and confirmed it doesn't work?

**Comment by jpSimkins** [comment]
@skinny85 Thank you for your response. 

To answer your question, I did not test this out. I assumed due to my experiences with Terraform and the need to associate the OAuth arn to the project. I created my codeBuild project with the CDK and when I inspected it, it was not connected. All I have to do was click the OAuth login and that solved the issue. This is a bit embarrassing... 

I apologize for opening this ticket without properly testing thoroughly. 
Thanks for your time.
